cr.c

```diff
@@ -84,9 +84,11 @@ static void dill_ctx_timer_handler(int sig, siginfo_t *si, void *uc) {
 #ifndef __APPLE__
     struct dill_ctx_cr *ctx = si->si_value.sival_ptr;
     int or = timer_getoverrun(ctx->timer);
-    if(or) ctx->do_poll = 1;
+    if(or) {
+        __sync_lock_test_and_set(&ctx->do_poll, 1);
+    }
 #else
-    dill_poll_count++;
+    __sync_fetch_and_add(&dill_poll_count, 1);
```

Comment:
Well, actually we can use a constant, assuming that frequency of the processor is something like 1GHz.  On 3GHz processor we will have to do clock_gettime() 3x as often, but it'll still be 3 times per millisecond, which should be statistically negligible.